src/rules/convertRules.test.ts

```diff
@@ -183,6 +183,96 @@ describe("convertRules", () => {
         );
     });
 
+    it("merges and deduplicates rule notices", () => {
+        // Arrange
+        const tslintRule: TSLintRuleOptions = {
+            ruleArguments: [],
+            ruleName: "tslint-rule-a",
+            ruleSeverity: "error",
+        };
+        const conversionResult = {
+            rules: [
+                {
+                    ruleName: "eslint-rule-a",
+                    notices: ["notice-1", "notice-2"],
+                },
+                {
+                    ruleName: "eslint-rule-a",
+                    notices: ["notice-1"],
+                },
+            ],
+        };
+        const mergedArguments = [{ merged: true }];
+        const converters = new Map([[tslintRule.ruleName, () => conversionResult]]);
+        const mergers = new Map([[conversionResult.rules[0].ruleName, () => mergedArguments]]);
+
+        // Act
+        const { converted } = convertRules(
+            { converters, mergers },
+            { [tslintRule.ruleName]: tslintRule },
+        );
+
+        // Assert
+        expect(converted).toEqual(
+            new Map([
+                [
+                    "eslint-rule-a",
+                    {
+                        ruleArguments: mergedArguments,
+                        ruleName: "eslint-rule-a",
+                        ruleSeverity: "error",
+                        notices: ["notice-1", "notice-2"],
+                    },
+                ],
+            ]),
+        );
+    });
+
+    it("merges undefined notices", () => {
+        // Arrange
+        const tslintRule: TSLintRuleOptions = {
+            ruleArguments: [],
+            ruleName: "tslint-rule-a",
+            ruleSeverity: "error",
+        };
+        const conversionResult = {
+            rules: [
+                {
+                    ruleName: "eslint-rule-a",
+                    notices: undefined,
+                },
+                {
+                    ruleName: "eslint-rule-a",
+                    notices: undefined,
```

Comment:
@JoshuaKGoldberg is this the test case you're looking for? (Will make a follow-up PR to DRY out these tests)

Reply:
Yeah, this is perfect! 🙌